common/utils/retry.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+func Retry(timeout, delay time.Duration, fn func() (bool, error)) error {
+	ticker := time.NewTicker(delay)
+	errc := make(chan error)
+	defer func() {
+		ticker.Stop()
+		close(errc)
+	}()
+
+	go func() { errc <- retry(ticker, fn) }()
+
+	select {
+	case err := <-errc:
+		return err
+	case <-time.After(timeout):
+		return errors.New(errors.FailedRequestTimeout, nil)
+	}
+}
+
+func retry(ticker *time.Ticker, fn func() (bool, error)) error {
+	for ; true; <-ticker.C {
```

Comment:
I believe this is a memory leak. I think <-ticker.C will hang forever after the channel is closed.